lib/pbench/server/sync.py

```diff
@@ -74,7 +73,12 @@ def next(self, operation: Operation) -> List[Dataset]:
             if self.logger.isEnabledFor(DEBUG):
                 q_str = query.statement.compile(compile_kwargs={"literal_binds": True})
                 self.logger.debug("QUERY {}", q_str)
-            return list(query.all())
+            datasets_l = []
+            for dataset in query.all():
+                op_val = Metadata.getvalue(dataset, Metadata.OPERATION)
+                if op_val and op_val[0] == operation.name:
```

Comment:
Should be `if op_val and operation.name in op_val` ... it might not be first.

(It'll probably work eventually, but that's not how it was intended to work. E.g., we may have `["backup", "index"]` and you're blocking index until backup is finished rather than letting them go in parallel. Since a "correct" fix isn't hard, I'd prefer to go that way.)

Reply:
Fixed.